Marlin/src/gcode/calibrate/G425.cpp

```diff
@@ -0,0 +1,673 @@
+ * and their destinations can be specified, in any order.
+ */
+inline void move_to(
+  const AxisEnum a1 = NO_AXIS, const float p1 = 0,
+  const AxisEnum a2 = NO_AXIS, const float p2 = 0,
+  const AxisEnum a3 = NO_AXIS, const float p3 = 0
+) {
+  set_destination_from_current();
+
+  // Note: The order of p1, p2, p3 may not correspond to X, Y, Z
+  if (a1 != NO_AXIS) destination[a1] = p1;
+  if (a2 != NO_AXIS) destination[a2] = p2;
+  if (a3 != NO_AXIS) destination[a3] = p3;
+
+  // Make sure coordinates are within bounds
+  destination[X_AXIS] = MAX(MIN(destination[X_AXIS], X_MAX_POS), X_MIN_POS);
+  destination[Y_AXIS] = MAX(MIN(destination[Y_AXIS], Y_MAX_POS), Y_MIN_POS);
+  destination[Z_AXIS] = MAX(MIN(destination[Z_AXIS], Z_MAX_POS), Z_MIN_POS);
+
+  // Move to position
+  do_blocking_move_to(destination[X_AXIS], destination[Y_AXIS], destination[Z_AXIS], MMM_TO_MMS(CALIBRATION_FEEDRATE_TRAVEL));
+}
+
+/**
+ * Move to the exact center above the calibration cube
+ *
+ *   m                  in     - Measurement record
+ *   uncertainty        in     - How far away from the cube top to park
+ */
+inline void park_above_cube(measurements_t &m, const float uncertainty) {
+  /* Move to safe distance above cube */
+  move_to(Z_AXIS, m.cube_center[Z_AXIS] + m.dimensions[Z_AXIS] / 2 + uncertainty);
+
+  /* Move to center of cube in XY */
+  move_to(X_AXIS, m.cube_center[X_AXIS], Y_AXIS, m.cube_center[Y_AXIS]);
+}
+
+#if HOTENDS > 1
+
+  inline void set_nozzle(measurements_t &m, const uint8_t extruder) {
+    if (extruder != active_extruder) {
+      park_above_cube(m, CALIBRATION_MEASUREMENT_UNKNOWN);
+      tool_change(extruder);
+    }
+  }
+
+  inline void reset_nozzle_offsets() {
+    constexpr float tmp[XYZ][HOTENDS] = { HOTEND_OFFSET_X, HOTEND_OFFSET_Y, HOTEND_OFFSET_Z };
+    LOOP_XYZ(i) HOTEND_LOOP() hotend_offset[i][e] = tmp[i][e];
+  }
+
+  inline void normalize_hotend_offsets() {
+    for (uint8_t e = 1; e < HOTENDS; e++) {
+      hotend_offset[X_AXIS][e] -= hotend_offset[X_AXIS][0];
+      hotend_offset[Y_AXIS][e] -= hotend_offset[Y_AXIS][0];
+      hotend_offset[Z_AXIS][e] -= hotend_offset[Z_AXIS][0];
+    }
+    hotend_offset[X_AXIS][0] = 0;
+    hotend_offset[Y_AXIS][0] = 0;
+    hotend_offset[Z_AXIS][0] = 0;
+  }
+
+  //
+  // This function requires normalize_hotend_offsets() to be called
+  //
+  inline void report_hotend_offsets() {
+    for (uint8_t e = 1; e < HOTENDS; e++) {
+      SERIAL_ECHOPAIR("T", int(e));
+      SERIAL_ECHOLNPGM(" Hotend Offset:");
+      SERIAL_ECHOLNPAIR("  X: ", hotend_offset[X_AXIS][e]);
+      SERIAL_ECHOLNPAIR("  Y: ", hotend_offset[Y_AXIS][e]);
+      SERIAL_ECHOLNPAIR("  Z: ", hotend_offset[Z_AXIS][e]);
+      SERIAL_EOL();
+    }
+  }
+
+#endif // HOTENDS > 1
+
+inline bool read_probe_value() {
+  #if ENABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN)
+    return (READ(Z_MIN_PIN) != Z_MIN_ENDSTOP_INVERTING);
+  #else
+    return (READ(Z_MIN_PROBE_PIN != Z_MIN_PROBE_ENDSTOP_INVERTING);
```

Comment:
Issue here. Needs a closing bracket on 193. 
`return (READ(Z_MIN_PROBE_PIN != Z_MIN_PROBE_ENDSTOP_INVERTING);`
should be
`return (READ(Z_MIN_PROBE_PIN) != Z_MIN_PROBE_ENDSTOP_INVERTING);`

Reply:
@SJ-Innovation: Good catch!